packages/core/migrations/1687164785494_init.mjs

```diff
@@ -0,0 +1,256 @@
+    .addColumn('source', 'jsonb', (col) => col.notNull())
+    .addColumn('inserted', 'timestamp', (col) => col.defaultTo(sql`timezone('utc', now())`))
+    .execute()
+  
+  /**
+   * Table describes pieces derived corresponding to CARs in the content table. Link (commP) is
+   * unique even though cargo reference is not, that is because there may be an error in piece
+   * derivation and in that case another correct piece will reference the same content.
+   */
+  await db.schema
+    .createTable('piece')
+    .addColumn('link', 'text', (col) => col.primaryKey())
+    .addColumn('size', 'bigint', (col) => col.notNull())
+    .addColumn('content', 'text', (col) => col.references('content.link'))
+    .addColumn('inserted', 'timestamp', (col) => col.defaultTo(sql`timezone('utc', now())`))
+    .execute()
+  
+  await db.schema
+    .createIndex('piece_content_idx')
+    .on('piece')
+    .column('content')
+    .execute()
+  await db.schema
+    .createIndex('piece_inserted_idx')
+    .on('piece')
+    .column('inserted')
+    .execute()
+  
+  /**
+   * Content for which we need to derive piece CIDs. We will have a process that
+   * reads from this queue and writes into `piece` table.
+   */
+  await db.schema
+    .createView('content_queue')
+    .as(
+      db.selectFrom('content')
+        .leftJoin('piece', 'content.link', 'piece.content')
+        .where('piece.content', 'is', null)
+        .select([
+          'content.link',
+          'content.size',
+          'content.source',
+          'content.inserted',
+        ])
+        .orderBy('piece.inserted')
+    )
+    .execute()
+  
+  /**
+   * Table for created aggregates.
+   */
+  await db.schema
+    .createTable('aggregate')
+    .addColumn('link', 'text', (col) => col.primaryKey())
+    .addColumn('size', 'bigint', (col) => col.notNull())
+    .addColumn('inserted', 'timestamp', (col) => col.defaultTo(sql`timezone('utc', now())`))
+    .execute()
+
+  /**
+   * Table describing pieces to be included into aggregates. If aggregate is NULL then the
+   * piece is queued for the aggregation.
+   */
+  await db.schema
+    .createTable('inclusion')
+    .addColumn('piece', 'text', (col) => col.references('piece.link').notNull())
+    .addColumn('aggregate', 'text', (col) => col.defaultTo(null))
+    .addColumn('priority', 'integer', (col) => col.notNull())
+    .addColumn('inserted', 'timestamp', (col) => col.defaultTo(sql`timezone('utc', now())`))
+    .addForeignKeyConstraint(
+      'aggregate_id_foreign',
+      ['aggregate'],
+      'aggregate',
+      ['link']
+    )
+    .execute()
+
+  await db.schema
+    .createIndex('inclusion_priority_idx')
+    .on('inclusion')
+    .expression(sql`priority DESC, inserted`)
+    .execute()
+
+    await db.schema
+      .createIndex('piece_aggregate_unique_idx')
+      .on('inclusion')
+      .unique()
+      // coalesce is used to guarantee uniqueness with null
+      .expression(sql`piece, COALESCE(aggregate, '0')`)
+      .execute()
```

Comment:
@Gozala note this. Looks like NULL is considered different value and unique constraints do not apply to it 😅  
I spent a lot of time trying to find a workaround this, given there are a few options but most of them are on the newest postgres versions that did not make it to AWS yet. See https://dba.stackexchange.com/questions/299098/why-doesnt-my-unique-constraint-trigger/299107#299107